arangod/Aql/MaterializeExecutor.h

```diff
@@ -88,6 +88,7 @@ class MaterializeExecutor {
   struct Properties {
     static constexpr bool preservesOrder = true;
     static constexpr BlockPassthrough allowsBlockPassthrough = BlockPassthrough::Disable;
+    // TODO this could be set to true!
```

Comment:
As I remembered it, `expectedNumberOfRows` had then to be implemented as well, or otherwise it failed to compile. I just looked it up in order to answer correctly, though. And as it happens, we now have `expectedNumberOfRowsNew` (okay, fine). And to my dismay I found out that `inputSizeRestrictsOutputSize` seems to just be ignored (unless I overlooked something), and the "switch" whether `expectedNumberOfRowsNew` is used is just dependent on the existence of a method with that name in the executor. Which I don't like at all, because now a typo is the difference between it working or not.

So I take away that we should to the following refactor steps:
- Change `expectedNumberOfRowsNew` to `expectedNumberOfRows`
- Change the name `inputSizeRestrictsOutputSize` to something more true, e.g. `canEstimateNumberOfOutputRows`
- Add an assertion in `ExecutionBlockImpl` that `Executor::canEstimateNumberOfOutputRows == hasExpectedNumberOfRowsNew<Executor>::value`

And, somewhat orthogonally to that, write an implementation of `expectedNumberOfRows` for `MaterializeExecutor`.

But none of that belongs into this PR.

Reply:
👍 sorry for being awkward